internal/services/directoryroles/directory_roles_data_source.go

```diff
@@ -0,0 +1,109 @@
+package directoryroles
+
+import (
+	"context"
+	"crypto/sha1"
+	"encoding/base64"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-azuread/internal/clients"
+	"github.com/hashicorp/terraform-provider-azuread/internal/tf"
+)
+
+func directoryRoleDataSource() *schema.Resource {
```

Comment:
We should keep the pluralization consistent

```suggestion
func directoryRolesDataSource() *schema.Resource {
```